// Copyright 2022 The SiliFuzz Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package silifuzz.proto;

import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";
import "proto/session_summary.proto";
import "proto/snapshot_execution_result.proto";

// A union of all message types that can be sent via a binary log channel.
// NextID: 7
message BinaryLogEntry {
  // ID of the session this entry belongs to.
  string session_id = 5;

  // Event timestamp.
  google.protobuf.Timestamp timestamp = 6;

  oneof kind {
    // Result of executing a snapshot
    SnapshotExecutionResult snapshot_execution_result = 1;

    // Build information of the producer binary.
    google.protobuf.Any build_info = 3;

    // Session summary.
    silifuzz.proto.logging.SessionSummary session_summary = 4;
  }
}
